crates/uv-python/src/downloads.rs

```diff
@@ -420,46 +424,63 @@ impl ManagedPythonDownload {
         // Ensure the request was successful.
         response.error_for_status_ref()?;
 
+        let size = response.content_length();
+        let progress = reporter
+            .as_ref()
+            .map(|reporter| (reporter, reporter.on_download_start(&self.key, size)));
+
         // Download and extract into a temporary directory.
         let temp_dir = tempfile::tempdir_in(parent_path).map_err(Error::DownloadDirError)?;
 
         debug!(
             "Downloading {url} to temporary location {}",
             temp_dir.path().display()
         );
-        let reader = response
+
+        let stream = response
             .bytes_stream()
             .map_err(|err| io::Error::new(io::ErrorKind::Other, err))
             .into_async_read();
 
+        let mut hashers = self
+            .sha256
+            .into_iter()
+            .map(|_| Hasher::from(HashAlgorithm::Sha256))
+            .collect::<Vec<_>>();
+        let mut hasher = uv_extract::hash::HashReader::new(stream.compat(), &mut hashers);
```

Comment:
I was kinda hoping to avoid hashing when `self.sha256` is `None`, any chance we can preserve that behavior?

Reply:
If `self.sha256` is `None`, `hashers` will be empty, and `HashReader` won't do any hashing. So I think the overhead might be negligible. 
https://github.com/astral-sh/uv/blob/1bd73a7346ca62e3a7440f32980fa59591535610/crates/uv-extract/src/hash.rs#L138-L140